src/RobotController.cpp

```diff
@@ -6,23 +6,45 @@
  */
 
 #include "RobotController.h"
-#include "AutoController.h"
 
-RobotController::RobotController(DriveStation* driveStation, AutoController* autoC) :
-   m_driveStation(driveStation), m_auto(autoC)
-{}
+RobotController::RobotController(DriveStation* ds, DriveTrainController* dt, ShooterController* shooter, LoaderController* loader)
+   : m_dstation(ds), m_drivet(dt), m_shooter(shooter), m_loader(loader)
+{
+   m_state = ROBOT_AUTO;
+}
+
+RobotController::~RobotController() {}
 
 void RobotController::run()
 {
-   // TODO: Put functionality for both manual and auto, not just autonomous
-   m_auto->run();
+   if (m_state == ROBOT_AUTO)
+      performAction();
```

Comment:
Does `performAction()` call the `run()` functions for the DriveTrain, Shooter, and Loader